docs/root/configuration/http/http_filters/header_to_metadata_filter.rst

```diff
@@ -39,6 +40,28 @@ absence of a version header could be:
               type: STRING
             remove: false
 
+As with headers, the value of the specified cookie will be extracted from the request
+and added as metdata with the key specified.
+
+.. code-block:: yaml
+
+  http_filters:
+    - name: envoy.filters.http.header_to_metadata
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.filters.http.header_to_metadata.v3.Config
+        request_rules:
+          - cookie: cookie
+            on_header_present:
+              metadata_namespace: envoy.lb
+              key: version
+              type: STRING
+            on_header_missing:
+              metadata_namespace: envoy.lb
+              key: default
+              value: 'true'
+              type: STRING
+            remove: false
```

Comment:
hmm it feels weird to use `on_header_present` and on `on_header_missing` to act on a cookie... Should we use `(udpa.annotations.field_migrate).rename = "on_present"` (and `on_missing`) to have these renamed in v4?

Reply:
cc: @zuercher

Reply:
I think it makes sense to have those 2 fields renamed. let me know what you both prefer, happy to make that change :)

Reply:
Yeah I like the idea of the renaming for v4. But we can do it in a follow-up PR too, to avoid delaying this one.